spec/observables/race-spec.js

```diff
@@ -0,0 +1,176 @@
+
+    expectObservable(result).toBe(expected);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should race cold and cold', function () {
+    var e1 =  cold('---a-----b-----c----|');
+    var e1subs =   '^                   !';
+    var e2 =  cold('------x-----y-----z----|');
+    var e2subs =   '^                      !';
+    var expected = '---a-----b-----c-------|';
+
+    var result = Observable.race(e1, e2);
+
+    expectObservable(result).toBe(expected);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+  });
+
+  it('should race hot and hot', function () {
+    var e1 =   hot('---a-----b-----c----|');
+    var e1subs =   '^                   !';
+    var e2 =   hot('------x-----y-----z----|');
+    var e2subs =   '^                      !';
+    var expected = '---a-----b-----c-------|';
+
+    var result = Observable.race(e1, e2);
+
+    expectObservable(result).toBe(expected);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+  });
+
+  it('should race hot and cold', function () {
+    var e1 =  cold('---a-----b-----c----|');
+    var e1subs =   '^                   !';
+    var e2 =   hot('------x-----y-----z----|');
+    var e2subs =   '^                      !';
+    var expected = '---a-----b-----c-------|';
+
+    var result = Observable.race(e1, e2);
+
+    expectObservable(result).toBe(expected);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+  });
+
+  it('should race emit and complete', function () {
+    var e1 =  cold('-----|');
+    var e1subs =   '^    !';
+    var e2 =   hot('------x-----y-----z----|');
+    var e2subs =   '^    !';
+    var expected = '-----|';
+
+    var result = Observable.race(e1, e2);
+
+    expectObservable(result).toBe(expected);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+  });
+
+  it('should allow unsubscribing early and explicitly', function () {
+    var e1 =  cold('---a-----b-----c----|');
+    var e1subs =   '^           !';
+    var e2 =   hot('------x-----y-----z----|');
+    var e2subs =   '^           !';
+    var expected = '---a-----b---';
+    var unsub =    '            !';
+
+    var result = Observable.race(e1, e2);
+
+    expectObservable(result, unsub).toBe(expected);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+  });
+
+  it('should not break unsubscription chains when unsubscribed explicitly', function () {
+    var e1 =  cold('---a-----b-----c----|');
+    var e1subs =   '^           !';
+    var e2 =   hot('------x-----y-----z----|');
+    var e2subs =   '^           !';
+    var expected = '---a-----b---';
+    var unsub =    '            !';
+
+    var result = Observable.race(e1, e2);
```

Comment:
This test should not be just a copy of the previous. You need to wrap e1 and e2 in a "mergeMap" sandwich, and also put a mergeMap after race. The idea is that unsubscribing from a mergeMap operator should send a message to race operator to unsubscribe which should send a message to a mergeMap operator to unsubscribe which in turn should send a message to e1 and e2 to unsubscribe.